### ISSUE TITLE
`Label Set Empty` Warning Messages

### ISSUE THREAD

**Issue by amogkam** [ray-project/xgboost_ray#12]
I am running simple.py with 200 workers, 1 CPU per worker. Training is able to complete, but the stdout is overwhelmed with a ton of these warning messages:

```
(pid=461, ip=172.31.23.214) [10:01:54] WARNING: ../src/metric/elementwise_metric.cu:336: label set is empty
(pid=11317) [10:01:54] WARNING: ../src/metric/elementwise_metric.cu:336: label set is empty
(pid=11317) [10:01:54] WARNING: ../src/metric/elementwise_metric.cu:336: label set is empty
(pid=313, ip=172.31.23.78) [10:01:54] WARNING: ../src/metric/elementwise_metric.cu:336: label set is empty
```
I did not see this warning message when I tried with 100 actors.

Here is the YAML file that I am using:
```
cluster_name: amog_xgboost_scale
min_workers: 100
max_workers: 200
initial_workers: 100
autoscaling_mode: default
docker:
    image: 'anyscale/ray-ml:latest'
    container_name: ray_container
    pull_before_run: false
    run_options:
        - --privileged
        - --shm-size=510000000000
target_utilization_fraction: 0.8
idle_timeout_minutes: 5
provider:
    type: aws
    region: us-west-2
    availability_zone: us-west-2a
    cache_stopped_nodes: false
auth:
    ssh_user: ubuntu
    ssh_private_key: ~/.ssh/amog-anyscale.pem
head_node:
    InstanceType: r5.16xlarge
    ImageId: ami-05ac7a76b4c679a79
    KeyName: amog-anyscale
worker_nodes:
    InstanceType: m5.xlarge
    ImageId: ami-05ac7a76b4c679a79
    KeyName: amog-anyscale
    InstanceMarketOptions:
        MarketType: spot
        
initialization_commands: []
setup_commands:
    - echo "hello"
    - pip install -U https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-1.1.0.dev0-cp37-cp37m-manylinux2014_x86_64.whl
    - pip install dask distributed xgboost s3fs
head_setup_commands: 
    - pip install -U git+https://github.com/ray-project/xgboost_ray@master#xgboost_ray
worker_setup_commands:
    - pip install -U git+https://github.com/ray-project/xgboost_ray@master#xgboost_ray
head_start_ray_commands:
    - ray stop
    - "ulimit -n 65536; ray start --head --port=6379 --object-manager-port=8076 --autoscaling-config=~/ray_bootstrap_config.yaml --object-store-memory=500000000000 --resources='{\"actor_cpus\": 0}'"
worker_start_ray_commands:
    - ray stop
    - "ulimit -n 65536; ray start --address=$RAY_HEAD_IP:6379 --object-manager-port=8076 --resources='{\"actor_cpus\": 4}'"
```

**Comment by richardliaw** [comment]
For these issues, can you be sure to post the repro command (num workers, shards, etc)?

**Comment by amogkam** [comment]
One thing I forgot to add- I am running 100 boosting rounds (instead of the default 10). As mentioned in the original post, I am running simple.py with 200 workers. simple.py does not do any distributed data loading.

**Comment by amogkam** [comment]
I can reproduce this on the latest master.

**Comment by krfricke** [comment]
This message changed to

```
(pid=308, ip=172.31.24.18) [05:20:46] WARNING: ../src/learner.cc:1190: Empty dataset at worker: 68
```

I suspect this is due to the test set. The breast cancer dataset has 569 rows. A split of 25% means 143 rows are in the test set, so some actors might have empty data. I just confirmed that the threshold for this message seems to be at 144 workers.

I suggest we throw an error if more actors are requested than the dataset can be split into. This has to happen on a per-matrix basis. I'll make a suggestion on where to introduce this check.
